src/functions/function_utils.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * MRCPP, a numerical library based on multiresolution analysis and
+ * the multiwavelet basis which provide low-scaling algorithms as well as
+ * rigorous error control in numerical computations.
+ * Copyright (C) 2019 Stig Rune Jensen, Jonas Juselius, Luca Frediani and contributors.
+ *
+ * This file is part of MRCPP.
+ *
+ * MRCPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MRCPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+#include "function_utils.h"
+#include "utils/details.h"
+
+#include "GaussExp.h"
+#include "Gaussian.h"
+#include "utils/Printer.h"
+
+#include <memory>
+
+namespace mrcpp {
+
+/** @brief Generates a GaussExp that is semi-periodic around a unit-cell
+ *
+ * @param[out] shared_ptr<GaussExp<D>> semi-periodic version of a Gaussian around a unit-cell
+ * @param[in] A Gaussian function that is to be made semi-periodic
+ * @param[in] The period of the unit cell
+ *
+ * The Gaussian function replicated in neighbooring cells, such that atleast
+ * 99.99367% if the integral can be captured within the unit-cell.
+ */
+template <int D>
+std::shared_ptr<GaussExp<D>> function_utils::make_gaussian_periodic(Gaussian<D> &inp,
+                                                                    const std::array<double, D> &period) {
+    auto gauss_exp = std::make_shared<GaussExp<D>>();
+    auto pos_vec = std::vector<Coord<D>>();
+
+    // Four standard deviation upp and down from the expectation value of the
+    // gaussian holds approximately 99.99367% of the integral.
+    auto four_std = 4.0 * inp.getMaximumStandardDiviation();
```

Comment:
Perhaps add `nStdDev` to the argument list of this function? Makes it a bit less hard coded